vulcanai2/models/basenetwork.py

```diff
@@ -215,22 +219,22 @@ def criter_spec(self, value):
 
     def get_flattened_size(self, network):
         """
-        Returns the flattened output size of the conv network's last layer.
+        Returns the flattened output size of a Single Input ConvNet's last layer.
         :param network: The network to flatten
         :return: The flattened output size of the conv network's last layer.
         """
         with torch.no_grad():
-            x = torch.ones(1, *self.in_dim)
+            x = torch.ones(1, *self.in_dim[0])
             x = network(x)
             return x.numel()
 
     def get_output_size(self):
         """
-        Returns the output size of the network's last layer
+        Returns the output size of the Single Input DenseNet's last layer
         :return: The output size of the network's last layer
         """
         with torch.no_grad():
-            x = torch.ones(1, self.in_dim)
+            x = torch.ones(1, self.in_dim[0])
```

Comment:
same comment as above on index-based retrieval

Reply:
Same - "works only for Single Input case"